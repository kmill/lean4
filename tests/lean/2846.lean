/-!
# `delabConstWithSignature` avoids using inaccessible names
-/

/-!
Defined without named arguments, prints without named arguments.
-/
#check String.append

/-!
The List argument is not named, it is not printed as a named argument.
-/
#check List.length

/-!
All arguments are named, all are printed as named arguments.
-/
#check Nat.pos_pow_of_pos

/-!
The hypothesis is not a named argument, so it's not printed as a named argument.
-/
def Nat.pos_pow_of_pos' {n : Nat} (m : Nat) : 0 < n â†’ 0 < n ^ m := Nat.pos_pow_of_pos m

#check Nat.pos_pow_of_pos'

/-!
Repetition of a named argument, only the first is printed as a named argument.
-/
def foo (n n : Nat) : Fin (n + 1) := 0

#check foo

/-!
Repetition of a named argument, only the first is printed as a named argument.
This is checking that shadowing is represented correctly (that's not the responsibility of
`delabConstWithSignature`, but it assumes that the main delaborator will handle this correctly).
-/
def foo' (n n : Nat) (a : Fin ((by clear n; exact n) + 1)) : Fin (n + 1) := 0

#check foo'
