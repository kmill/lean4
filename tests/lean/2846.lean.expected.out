String.append : String → String → String
List.length.{u_1} {α : Type u_1} : List α → Nat
Nat.pos_pow_of_pos {n : Nat} (m : Nat) (h : 0 < n) : 0 < n ^ m
Nat.pos_pow_of_pos' {n : Nat} (m : Nat) : 0 < n → 0 < n ^ m
foo (n : Nat) : (n : Nat) → Fin (n + 1)
foo' (n : Nat) : (n_1 : Nat) → Fin (n + 1) → Fin (n_1 + 1)
foo'' : String → String → String
bar : String → (n : Nat) → Fin (n + 1)
2846.lean:62:4-62:8: warning: declaration uses 'sorry'
bar'.{u_1} {α : Type u_1} [LT α] (x : α) : x < x
